### PR TITLE
Support added for Segment routing,policy-class map,pce,mpls_ldpconfig

### PR DESCRIPTION
**Pull request by swarudka**

### CODE REVIEW

File: gen/definitions/qos_class_map.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+doc_category: QoS
+attributes:
+  - yang_name: class-map-name
+    id: true
+    example: TEST
+  - yang_name: match-any
+    example: true
+  - yang_name: description
+    example: description1
+  - yang_name: match/dscp/value
```
Comment (danischm) [inline gen/definitions/qos_class_map.yaml:13]: `tf_name: match_dscp`

File: gen/definitions/qos_class_map.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    example: 46
+  - yang_name: match/mpls/experimental/topmost/label
```
Comment (danischm) [inline gen/definitions/qos_class_map.yaml:15]: `tf_name: match_mpls_experimental_topmost`

File: gen/definitions/qos_class_map.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    example: 5
+  - yang_name: match/qos-group/id
```
Comment (danischm) [inline gen/definitions/qos_class_map.yaml:17]: `tf_name: match_qos_group`

File: gen/definitions/qos_class_map.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    example: group1
```
Comment (danischm) [inline gen/definitions/qos_class_map.yaml:18]: I think this should be "1".

File: gen/definitions/qos_class_map.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    exclude_test: true
+  - yang_name: match/traffic-class/id
```
Comment (danischm) [inline gen/definitions/qos_class_map.yaml:20]: `tf_name: match_traffic_class`

File: gen/definitions/qos_class_map.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    example: traffic1
```
Comment (danischm) [inline gen/definitions/qos_class_map.yaml:21]: this should be "1"

File: gen/definitions/qos_policy_map.yaml [hunk 7]
```diff
@@ -0,0 +1,52 @@
+---
+name: QOS Policy Map
+path: Cisco-IOS-XR-um-policymap-classmap-cfg:/policy-map/type/qos[policy-map-name=%s]
+doc_category: QoS
+attributes:
+  - yang_name: policy-map-name
+    id: true
+    example: core-ingress-classifier
+  - yang_name: class/name
```
Comment (danischm) [inline gen/definitions/qos_policy_map.yaml:9]: `class` is a list

File: gen/definitions/segment_routing_te.yaml [hunk 8]
```diff
@@ -3,10 +3,40 @@ name: Segment Routing TE
...
+        id: true 
+        example: 66.66.66.6
+      - yang_name: precedence
+        example: 122
+        
   - yang_name: on-demand-colors/on-demand-color
     tf_name: on_demand_colors
     type: List
     attributes: 
+      - yang_name: on-demand-color-dyn-mpls/on-demand-color-dyn-mpls-anycast
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:36]: `tf_name: dynamic_anycast_sid_inclusion`

File: gen/definitions/segment_routing_te.yaml [hunk 9]
```diff
@@ -3,10 +3,40 @@ name: Segment Routing TE
...
+      - yang_name: precedence
+        example: 122
+        
   - yang_name: on-demand-colors/on-demand-color
     tf_name: on_demand_colors
     type: List
     attributes: 
+      - yang_name: on-demand-color-dyn-mpls/on-demand-color-dyn-mpls-anycast  
+        example: true
+      - yang_name: on-demand-color-dyn-mpls/on-demand-color-dyn-mpls-metric/metric-type
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:38]: `tf_name: dynamic_metric_type`

File: gen/definitions/segment_routing_te.yaml [hunk 10]
```diff
@@ -3,10 +3,40 @@ name: Segment Routing TE
...
+    example: true     
+  - yang_name: logging/policy-status
+    example: true
+  - yang_name: pcc/report-all
+    example: true
+  - yang_name: pcc/source-address
+    example: 88.88.88.8
+  - yang_name: pcc/delegation-timeout
+    example: 10
+  - yang_name: pcc/dead-timer-interval
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:16]: `tf_name: pcc_dead_timer`

File: gen/definitions/segment_routing_te.yaml [hunk 11]
```diff
@@ -3,10 +3,40 @@ name: Segment Routing TE
...
+    example: 60
+  - yang_name: pcc/initiated-state-interval
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:18]: `tf_name: pcc_initiated_state`

File: gen/definitions/segment_routing_te.yaml [hunk 12]
```diff
@@ -3,10 +3,40 @@ name: Segment Routing TE
...
+    example: 15
+  - yang_name: pcc/initiated-orphan-interval
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:20]: `tf_name: pcc_initiated_orphan`

**Review by danischm — CHANGES_REQUESTED** [review]

File: gen/definitions/mpls_ldp.yaml [hunk 13]
```diff
@@ -19,3 +19,31 @@ attributes:
       - yang_name: interface-name
         id: true
         example: GigabitEthernet0/0/0/1
+  
+  - yang_name: capabilities/sac/ipv4-disable
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/mpls_ldp.yaml:23]: I dont see mldp-only in YANG suite (pls check attached pic)
YANG Model : Cisco-IOS-XR-um-mpls-ldp-cfg
image

[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/mpls_ldp.yaml:23]: pls suggest here

[the remark this answers is not in the document]
Comment (danischm) [reply gen/definitions/mpls_ldp.yaml:23]: Indeed, it looks like `capabilities sac mldp-only` is an alias to disable all 4 options. Then let's also add the other `sac` leafs.

**Review by danischm — CHANGES_REQUESTED** [review]

**Review by danischm — APPROVED** [review]

**Pull request merged**